common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -238,7 +244,7 @@ extern "C" {
 
 namespace impl = KokkosTools::Sampler;
 
-EXPOSE_TOOL_SETTINGS(kokkosp_request_tool_settings)
+EXPOSE_TOOL_SETTINGS(impl::kokkosp_request_tool_settings)
```

Comment:
This resulted in an infinite loop.

Reply:
@masterleinad Thanks for pointing this out! Do you have a particular code you tested this on that gave this result? 

By the way, I am currently creating test cases for all the tools connectors. (Also, from this, an idea I had is to create a generic tools utility and generic tools connector. This would address Kokkos Tools Github Issue #92 and make the development. ) This is a bigger effort but my focus is on the sampler. 

This fix you suggest should be part of a separate github PR.

Reply:
No, I didn't actually test it. This just gave compiler warnings (for this to result in an infinite loop which makes sense) and that's why I fix it here.

Reply:
OK, got it.

That makes sense to put the fix in this PR.

Thanks.